File: src/main/kotlin/com/joshlong/jobs/watchdog/WatchdogProperties.java
package com.joshlong.jobs.watchdog;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "watchdog")
public class WatchdogProperties {

	private long inactivityThresholdInSeconds = 5 * 60;

	public long getInactivityThresholdInSeconds() {
		return inactivityThresholdInSeconds;
	}

	public void setInactivityThresholdInSeconds(long inactivityThresholdInSeconds) {
		this.inactivityThresholdInSeconds = inactivityThresholdInSeconds;
	}

	public long getInactivityHeartbeatInSeconds() {
		return inactivityHeartbeatInSeconds;
	}

	public void setInactivityHeartbeatInSeconds(long inactivityHeartbeatInSeconds) {
		this.inactivityHeartbeatInSeconds = inactivityHeartbeatInSeconds;
	}

	private long inactivityHeartbeatInSeconds = 1;
}

File: src/main/kotlin/com/joshlong/jobs/watchdog/WatchdogProperties.kt
package com.joshlong.jobs.watchdog

import org.springframework.boot.context.properties.ConfigurationProperties

@ConfigurationProperties(prefix = "watchdog")
class WatchdogProperties(
	var inactivityThresholdInSeconds: Long = 5 * 60,
	var inactivityHeartbeatInSeconds: Long = 1
)